rs/nns/integration_tests/src/why_list_neurons_expensive.rs

```diff
@@ -0,0 +1,417 @@
+            GOVERNANCE_CANISTER_ID,
+            "where_ic_wasm_instrument_memory",
+            Encode!().unwrap(),
+        )),
+        u64,
+        u64
+    )
+    .unwrap();
+    let start_address = start_page * WASM_PAGE_SIZE_IN_BYTES as u64;
+    println!("");
+    println!("Result from ic_wasm_instrument_memory:");
+    println!(
+        "  start_page = {} (start_address = {})",
+        start_page, start_address
+    );
+    println!("  page_limit = {}", page_limit);
+    println!("");
+
+    // DO NOT MERGE: This is a hack that makes us use a only around half of the
+    // stable memory that was allocated for tracing/profiling.
+    let page_limit = page_limit / 2;
+
+    // Enable profiling of the governance canister.
+
+    // Have ic-wasm instrument the governance WASM for profiling.
+    let mut instrumented_governance_wasm =
+        walrus::Module::from_buffer(&decompress_gz(&governance_wasm_gz))
+            .expect("The Walrus library cannot cope with our WASM.");
+    ic_wasm::instrumentation::instrument(
+        &mut instrumented_governance_wasm,
+        ic_wasm::instrumentation::Config {
+            trace_only_funcs: vec![],
+            start_address: Some(i64::try_from(start_address).unwrap()),
+            page_limit: Some(i32::try_from(page_limit).unwrap()),
+        },
+    )
+    .unwrap();
+    let instrumented_governance_wasm = instrumented_governance_wasm.emit_wasm();
+
+    // Install the profiling-enabled governance WASM.
+    println!("");
+    println!("Installing governance WITH ic-wasm profiling ENABLED...");
+    println!("");
+    state_machine
+        .upgrade_canister(
+            GOVERNANCE_CANISTER_ID,
+            instrumented_governance_wasm.clone(),
+            vec![], // args
+        )
+        .unwrap();
+    println!("");
+    println!("Done installing governance WITH ic-wasm profiling ENABLED.");
+    println!("Ready for fine-grained performance measurement 👍");
+    println!("");
+
+    // Grab a sample of principals that have "associated" neurons. More
+    // precisely, these principals either control or are a hotkey of some
+    // (nonzero number of) neurons. These will later be used as callers of the
+    // list_neurons method.
+    let principal_id_to_neuron_count = state_machine
+        .execute_ingress(
+            GOVERNANCE_CANISTER_ID,
+            "principal_id_to_neuron_count",
+            Encode!().unwrap(),
+        )
+        .unwrap();
+    let principal_id_to_neuron_count = match principal_id_to_neuron_count {
+        ic_types::ingress::WasmResult::Reply(ok) => ok,
+        doh => panic!("{:?}", doh),
+    };
+    let principal_id_to_neuron_count = Decode!(
+        &principal_id_to_neuron_count,
+        Vec<(PrincipalId, (u64, u64))>
+    )
+    .unwrap();
+    println!("");
+    println!("principal_id_to_neuron_count:");
+    println!(
+        "{}",
+        clamp_debug_len(&principal_id_to_neuron_count, 1 << 10)
+    );
+    println!("");
```

Comment:
This is not of general interest to ppl who want to make their own flamegraphs. Rather, this is just to figure out what data to feed to the code that we are about to measure.